backend/routes/items.js

```diff
@@ -125,4 +125,37 @@ router.post("/addItem", token_required, async (req, res, next) => {
   });
 });
 
+router.put("/editItem", token_required, async (req, res, next) => {
+  // authentication
+  try {
+    // req.user is the user id fetched from Middleware
+    const user = await User.findById(req.user.id);
+  } catch (e) {
+    console.log(e);
+    return res.send({ message: "Error in Fetching user" }, 401);
+  }
+
+  try {
+    const { itemName, batchId, weight } = req.body;
+    const outDate = toUTCMidnight(new Date(req.body.outDate));
+    let batch = await Batch.findOne({ itemName, batchId });
+    
+    if (outDate < batch.inDate) {
+      return res.status(400).json({
+        fieldName: "outDate",
+        message: "Outdate is in the past",
+      });
+    }
+    if (weight > batch.poundsTotal) {
+      return res.status(400).json({ fieldName: "poundsRemaining", message: "Pounds remaining is greater than pounds total" })
+    }
+    
+    batch = await Batch.findOneAndUpdate({ itemName, batchId }, { outDate, poundsRemaining: weight });
```

Comment:
So I guess if the poundsTotal is bigger than it was before, we need to update that as well.

Reply:
The edit form on figma only has weight so I was following that. Should we add a field to edit the poundsTotal?